DocumentsFromSnapshotMigration/src/main/java/com/rfs/RfsMigrateDocuments.java

```diff
@@ -130,8 +132,19 @@ public static void main(String[] args) throws Exception {
             SnapshotShardUnpacker.Factory unpackerFactory = new SnapshotShardUnpacker.Factory(repoAccessor,
                     luceneDirPath, ElasticsearchConstants_ES_7_10.BUFFER_SIZE_IN_BYTES);
 
-            run(LuceneDocumentsReader::new, reindexer, workCoordinator, processManager, indexMetadataFactory,
-                    arguments.snapshotName, arguments.indexAllowlist, shardMetadataFactory, unpackerFactory, arguments.maxShardSizeBytes);
+            try {
+                run(LuceneDocumentsReader::new, reindexer, workCoordinator, processManager, indexMetadataFactory,
+                        arguments.snapshotName, arguments.indexAllowlist, shardMetadataFactory, unpackerFactory, arguments.maxShardSizeBytes);
+            } catch (NoWorkLeftException e) {
+                log.info("No work left to do.  Exiting.");
+                System.exit(0);
```

Comment:
Chris & I found why the main method returning wasn't enough for the process to die and this PR is superceded by https://github.com/opensearch-project/opensearch-migrations/pull/767